### PR TITLE
scripts: tests: twister_blackbox: Add more tests test_printouts.py

### PR DESCRIPTION
**Pull request by KamilxPaszkiet**
add tests to test_printouts.py:
  - timestamps
  - broken_parameter
  - help (dummy check)
  - force_color (dummy check)

### CODE REVIEW

File: scripts/tests/twister_blackbox/test_hardwaremap.py [hunk 1]
```diff
@@ -0,0 +1,318 @@
...
+                    assert load_data == expected_data
+
+                    if os.path.exists(path):
+                        os.remove(path)
+
+                    assert str(sys_exit.value) == '0'
+                    loggers = [logging.getLogger()] \
+                        + list(logging.Logger.manager.loggerDict.values()) \
+                        + [logging.getLogger(name) for \
+                                    name in logging.root.manager.loggerDict]
```
Comment (LukaszMrugala) [inline scripts/tests/twister_blackbox/test_hardwaremap.py:171]: Previously, we've had [plus] [newline], now it's [newline] [plus] - probably should be normalised.

File: scripts/tests/twister_blackbox/test_hardwaremap.py [hunk 2]
```diff
@@ -0,0 +1,318 @@
...
+                        f'  runner: {runner}\n'+\
+                        '  serial: /dev/ttyUSB24\n'+\
+                        '- connected: true\n'+\
+                        f'  id: {serial+3}\n'+\
+                        '  platform: unknown\n'+\
+                        f'  product: {product}\n'+\
+                        f'  runner: {runner}\n'+\
+                        '  serial: /dev/ttyUSB25\n'
+
+        load_data = open('test-map.yaml').read()
```
Comment (LukaszMrugala) [inline scripts/tests/twister_blackbox/test_hardwaremap.py:254]: See L161

File: scripts/tests/twister_blackbox/test_hardwaremap.py [hunk 3]
```diff
@@ -0,0 +1,318 @@
...
+
+        expected_data = '- connected: true\n'+\
+                        f'  id: {serial}\n'+\
+                        '  platform: unknown\n'+\
+                        f'  product: {product}\n'+\
+                        '  runner: pyocd\n'+\
+                        '  serial: /dev/ttyUSB23\n'
+        expected_data2 = '[]\n'
+
+        load_data = open('test-map.yaml').read()
```
Comment (LukaszMrugala) [inline scripts/tests/twister_blackbox/test_hardwaremap.py:310]: See L161

File: scripts/tests/twister_blackbox/test_printouts.py [hunk 4]
```diff
@@ -166,3 +175,41 @@ def test_tree(self, capfd, test_path, expected):
...
+
+        info_regex = r'\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2},\d{3} - (?:INFO|DEBUG|ERROR)'
+
+        out, err = capfd.readouterr()
+        sys.stdout.write(out)
+        sys.stderr.write(err)
+
+        output = err.split('\n')
+
+        err_lines=[]
```
Comment (LukaszMrugala) [inline scripts/tests/twister_blackbox/test_printouts.py:205]: Spaces around the equals sign

**Review by LukaszMrugala — DISMISSED** [review]
I don't see any explicit errors.
One safety question and one futureproofing change.
The rest is mostly style comments.

File: scripts/tests/twister_blackbox/test_hardwaremap.py [hunk 5]
```diff
@@ -0,0 +1,318 @@
...
+                    sys.stdout.write(out)
+                    sys.stderr.write(err)
+
+                    assert os.path.exists(path)
+
+                    expected_data = '- connected: true\n'+\
+                                    f'  id: {id_serial}\n'+\
+                                    '  platform: unknown\n'+\
+                                    f'  product: {id_pro}\n'+\
+                                    f'  runner: {runner}\n'+\
```
[the remark this answers is not in the document]
Comment (KamilxPaszkiet) [reply scripts/tests/twister_blackbox/test_hardwaremap.py]: fixed

File: scripts/tests/twister_blackbox/test_hardwaremap.py [hunk 6]
```diff
@@ -0,0 +1,318 @@
...
+            os.remove(path)
+
+        def mocked_comports():
+            return [
+                mock.Mock(device='/dev/ttyUSB23', \
+                    manufacturer= id_man, \
+                    product = id_pro, \
+                    serial_number = id_serial
+                    )
+                    ]
```
[the remark this answers is not in the document]
Comment (KamilxPaszkiet) [reply scripts/tests/twister_blackbox/test_hardwaremap.py]: just to make it easier to read

File: scripts/tests/twister_blackbox/test_hardwaremap.py (hunk 1 above)
Comment (KamilxPaszkiet) [reply scripts/tests/twister_blackbox/test_hardwaremap.py]: fixed!

File: scripts/tests/twister_blackbox/test_hardwaremap.py (hunk 2 above)
Comment (KamilxPaszkiet) [reply scripts/tests/twister_blackbox/test_hardwaremap.py]: fixed

File: scripts/tests/twister_blackbox/test_hardwaremap.py (hunk 3 above)
Comment (KamilxPaszkiet) [reply scripts/tests/twister_blackbox/test_hardwaremap.py]: fixed

File: scripts/tests/twister_blackbox/test_printouts.py (hunk 4 above)
Comment (KamilxPaszkiet) [reply scripts/tests/twister_blackbox/test_printouts.py]: fixed

Comment (LukaszMrugala) [reply scripts/tests/twister_blackbox/test_printouts.py]: I've meant `err_lines = []` rather than `err_lines=[ ]`

Comment (KamilxPaszkiet) [reply scripts/tests/twister_blackbox/test_printouts.py]: oh, ok... thx 😃

**Comment by nashif** [comment]
The help output test is just going to be a pain in the back, it does not really test any twister functionality and will fail whenever we added a new option or move things around or even change the help messages. I would remove it.

**Comment by KamilxPaszkiet** [comment]
> The help output test is just going to be a pain in the back, it does not really test any twister functionality and will fail whenever we added a new option or move things around or even change the help messages. I would remove it.

I deleted this and left the dummy test

**Review by LukaszMrugala — DISMISSED** [review]

**Review by nashif — APPROVED** [review]

**Review by LukaszMrugala — APPROVED** [review]
Minor nitpicks, I'd merge it.
`test_help` and `test_force_color` are check-for-errors only, but I do not see a nice way for them to be fully tested.

**Pull request merged**